src/mds/agg_mds/datastore/elasticsearch_dao.py

```diff
@@ -165,27 +262,84 @@ async def get_commons():
         return []
 
 
-async def get_all_metadata(limit, offset):
+def count(value) -> int:
+    """
+    returns the length of the value if list or dict otherwise returns 0
+    """
+    if isinstance(value, dict) or isinstance(value, list):
+        return len(value)
+    return 0
+
+
+def process_record(record: dict, counts: Optional[str]) -> Tuple[str, dict]:
+    """
+    processed an MDS record from the search
+    returns the id and record, if counts is found in the record the length is returned
+    instead of the entry.
+    """
+    _id = record["_id"]
+    normalized = record["_source"]
+    if counts in normalized:
+        normalized[counts] = count(normalized[counts])
+    return _id, normalized
+
+
+async def get_all_metadata(limit, offset, counts: Optional[str] = None, flatten=False):
+    """
+    Queries elastic search for metadata and returns up to the limit
+    offset: starting index to return
+    counts: converts the count of the entry[count] if it is a dict or array
+    returns:
+    flattend == true
+    results : MDS results as a dict
+              paging info
+    flattend == false
+    results : {
+        commonsA: metadata
+        commonsB: metadata
+        ...
+        },
+        paging info
+    """
     try:
         res = elastic_search_client.search(
             index=AGG_MDS_INDEX,
             body={"size": limit, "from": offset, "query": {"match_all": {}}},
         )
-        byCommons = {}
-        for record in res["hits"]["hits"]:
-            id = record["_id"]
-            normalized = record["_source"]
-            commons_name = normalized["commons_name"]
-            if commons_name not in byCommons:
-                byCommons[commons_name] = []
-            byCommons[commons_name].append(
-                {
-                    id: {
-                        "gen3_discovery": normalized,
-                    }
-                }
-            )
-        return byCommons
+        if flatten:
+            flat = []
+            for record in res["hits"]["hits"]:
+                id, normalized = process_record(record, counts)
+                flat.append({id: {"gen3_discovery": normalized}})
+            return {
+                "results": flat,
+                "pagination": {
+                    "hits": res["hits"]["total"],
```

Comment:
This is being called a lot, We should add this as a variable above instead of referencing the indices every time.

Reply:
Good call, it has been replaced with a variable as suggested.